### PR TITLE
Use Unicode codepoint count when validating string lengths

### LINKED ISSUE #10: Unicode support
came up in discussion of https://github.com/FiftyThree/SyncSchema/pull/49

- is JSON11 parsing unicode input as UTF-8, or other?
- length validation should count code points, not bytes

https://mathiasbynens.be/notes/javascript-unicode

```
>> '💩'.length // U+1F4A9 PILE OF POO
2

>> '💩' == '\uD83D\uDCA9'
true
```

### PR DESCRIPTION
**Pull request by floatplane**
Fixes #10.

Also another commit in here to blow away some old test files that aren't used by our fork.

PTAL @petersibley

### CODE REVIEW

File: src/tr/jsonschema/templates_cpp/static/UTF8Utils.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#include "UTF8Utils.h"
+
+using namespace std;
+
+size_t js2model::utf8_codepoint_count(const string &input) {
```
Comment (petersibley) [inline src/tr/jsonschema/templates_cpp/static/UTF8Utils.cpp]: interesting, odd that this isn't in stl but prolly not surprising given the mess of an API that `locales` is.

File: test/validation_tests.cpp [hunk 2]
```diff
@@ -50,7 +54,50 @@ TEST_CASE( "String validation" ) {
     }
 }
 
- TEST_CASE( "Integer validation" ) {
+static Json LoadEmojiTestData() {
+    ifstream data("jsonData/emoji-test.data.json");
```
Comment (petersibley) [inline test/validation_tests.cpp]: consider wrapping the `ifstream + stringstream + parse` dance into a util we've used it in a couple tests now.

**Comment by petersibley** [comment]
lgtm

File: src/tr/jsonschema/templates_cpp/static/UTF8Utils.cpp (hunk 1 above)
Comment (floatplane) [inline src/tr/jsonschema/templates_cpp/static/UTF8Utils.cpp]: Yeah, `std::codecvt::length` looked promising, but returns the number of UTF-8 bytes consumed to produce a certain number of wide codepoints. There probably is a way to do it, but this was very simple to write and definitely doesn't require converting the string to UCS-4.

File: test/validation_tests.cpp (hunk 2 above)
Comment (floatplane) [inline test/validation_tests.cpp]: Done.

**Pull request merged**